### PR TITLE
Enable model.eval() usage for `YOLO` class

### LINKED ISSUE #17751: models created by YOLO(..) can not be set to eval mode directly
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

The `YOLO` class is a subclass of `ultralytics.engine.Model`, and `ultralytics.engine.Model` is a subclass of `nn.Module`, so the model created by `YOLO`, for instance, `model = YOLO("yolo11n.pt")` can be set to `eval mode` logically.

However, when excute `model.eval()`, it launches training progress unexpectly, since the `YOLO` class overwrites `train()` function.

In concrete terms, `model.eval()` first called `eval()` function built in `nn.Module`, and here is the code:
```python
# In nn.Module
def eval(self: T) -> T:
        r"""Sets the module in evaluation mode.

        This has any effect only on certain modules. See documentations of
        particular modules for details of their behaviors in training/evaluation
        mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
        etc.

        This is equivalent with :meth:`self.train(False) <torch.nn.Module.train>`.

        See :ref:`locally-disable-grad-doc` for a comparison between
        `.eval()` and several similar mechanisms that may be confused with it.

        Returns:
            Module: self
        """
        return self.train(False)
```
It can be seen that `eval()` return `self.train(False)`, this line exactly aims to call the `train()` function defined in `nn.Module`, as for now, `YOLO` inherits `nn.Module` and overwrites `train()` function,  so the `eval()` function actually calls `YOLO: train()`, instead of `nn.Module: train()`, 

And the `YOLO: train()` is designed to train the model, so the operation of calling `model.eval()`, triggers  **training progress.**

### Environment

Ultralytics 8.3.29 🚀 Python-3.8.12 torch-1.11.0 CUDA:0 (NVIDIA GeForce RTX 4090, 24217MiB)
Setup complete ✅ (128 CPUs, 1007.5 GB RAM, 2085.6/28498.9 GB disk)

OS                  Linux-5.4.0-200-generic-x86_64-with-glibc2.17
Environment         Docker
Python              3.8.12
Install             git
RAM                 1007.53 GB
Disk                2085.6/28498.9 GB
CPU                 Intel Xeon Platinum 8369B 2.90GHz
CPU count           128
GPU                 NVIDIA GeForce RTX 4090, 24217MiB
GPU count           2
CUDA                11.3

numpy                    ✅ 1.24.4>=1.23.0
matplotlib               ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow                      ✅ 9.0.1>=7.1.2
pyyaml                    ✅ 6.0>=5.3.1
requests                  ✅ 2.28.2>=2.23.0
scipy                        ✅ 1.10.1>=1.4.1
torch                       ✅ 1.11.0>=1.8.0
torchvision              ✅ 0.12.0>=0.9.0
tqdm                       ✅ 4.65.2>=4.64.0
psutil                       ✅ 5.8.0
py-cpuinfo              ✅ 9.0.0
pandas                    ✅ 2.0.3>=1.1.4
seaborn                   ✅ 0.13.2>=0.11.0
ultralytics-thop       ✅ 2.0.11>=2.0.0
numpy                    ✅ 1.24.4<2.0.0; sys_platform == "darwin"
torch                       ✅ 1.11.0!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

```python
from ultralytics import YOLO

# Create minimal reproducible example
model = YOLO('yolov8n.pt')

print(f"Initial training state: {model.model.training}")

# Test eval()
model.eval()  # ---> here, triggers training progress
print(f"After eval(): {model.model.training}")
```

### Additional

Although `model.model.eval()` is useful, but maybe it is not elegant enough......

Based on the design of wrapper, `YOLO` has an attrbute named `model`, actually, what I want is setting this attribute to`eval mode`, so, we can add a function called `eval() `in `Model` class (the class that `YOLO` inherits from), so as to avoid calling `nn.Module: eval()`, and here is the code:
```python
class Model:
    ...
    def eval(self):
        self.model.eval()
        return self
```

 

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Laughing-q**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
A new `eval` method was added to set models to evaluation mode, which is crucial for consistent performance during model assessments.

### 📊 Key Changes
- Added an `eval` method to the `Model` class in `ultralytics/engine/model.py`.
- This method switches the model to evaluation mode, affecting components like dropout and batch normalization layers.

### 🎯 Purpose & Impact
- **Purpose**: To provide a straightforward way to switch the model to evaluation mode, ensuring layers operate consistently during inference.
- **Impact**: Users can now easily prepare their models for evaluation, potentially improving the accuracy and reliability of model assessments by avoiding training-specific behaviors. This enhances usability for both developers and non-expert users.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Laughing-q, thank you for submitting an `ultralytics/ultralytics` 🚀 pull request! To ensure a smooth review process and successful integration of your contribution, please check off the following steps:

- ✅ **Define a Purpose**: Please ensure your PR description clearly states the purpose of your changes. Consider linking any related [open issues](https://github.com/ultralytics/ultralytics/issues) to provide more context. Make sure your commit messages are precise and follow our project's conventions.
- ✅ **Synchronize with Source**: Make sure your branch is up-to-date with the `main` branch of `ultralytics/ultralytics`. If needed, you can update your branch by clicking 'Update branch' or running `git pull` and `git merge main` locally to resolve conflicts.
- ✅ **Ensure CI Checks Pass**: Verify that all necessary [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are successful. If any issues arise, please address them prior to merging.
- ✅ **Update Documentation**: If your PR introduces new features or changes existing ones, please update the appropriate [documentation](https://docs.ultralytics.com).
- ✅ **Add Tests**: Ensure that new or modified code is covered by unit tests, and confirm that all tests are passing as expected.
- ✅ **Sign the CLA**: If this is your first contribution to Ultralytics, please sign our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) by adding "I have read the CLA Document and I sign the CLA" in a new comment.
- ✅ **Minimize Changes**: Keep your changes focused and concise. "It is not daily increase but daily decrease, hack away the unessential.” — Bruce Lee

For detailed instructions, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). This is an automated response 😊, and an Ultralytics team member will review your PR soon. If you have any questions in the meantime, feel free to ask. Thank you for your contribution to Ultralytics! 🚀

**Comment by glenn-jocher** [comment]
@Laughing-q looks good!
